ydb/docs/en/core/integrations/ingestion/spark.md

```diff
@@ -0,0 +1,298 @@
+ |-- ViewCount: long (nullable = true)
+ |-- Body: binary (nullable = true)
+ |-- OwnerUserId: long (nullable = true)
+ |-- OwnerDisplayName: binary (nullable = true)
+ |-- LastEditorUserId: long (nullable = true)
+ |-- LastEditorDisplayName: binary (nullable = true)
+ |-- LastEditDate: timestamp (nullable = true)
+ |-- LastActivityDate: timestamp (nullable = true)
+ |-- Title: binary (nullable = true)
+ |-- Tags: binary (nullable = true)
+ |-- AnswerCount: long (nullable = true)
+ |-- CommentCount: long (nullable = true)
+ |-- FavoriteCount: long (nullable = true)
+ |-- ContentLicense: binary (nullable = true)
+ |-- ParentId: binary (nullable = true)
+ |-- CommunityOwnedDate: timestamp (nullable = true)
+ |-- ClosedDate: timestamp (nullable = true)
+
+scala> so_posts2020.count
+res1: Long = 4304021
+```
+
+Then add a new column with the year to this DataSet and store it all to a column-oriented {{ ydb-short-name }} table:
+
+```shell
+scala> val my_ydb = Map("url" -> "grpcs://ydb.my-host.net:2135/preprod/spark-test?tokenFile=~/.token")
+my_ydb: scala.collection.immutable.Map[String,String] = Map(url -> grpcs://ydb.my-host.net:2135/preprod/spark-test?tokenFile=~/.token)
+
+scala> so_posts2020.withColumn("Year", lit(2020)).write.format("ydb").options(my_ydb).option("table.type", "column").option("table.primary_keys", "Id").mode("append").save("stackoverflow/posts");
+```
+
+As a result, we can read the stored data from {{ ydb-short-name }} table and, for example, count the number of posts that have an accepted answer:
+
+```shell
+scala> val ydb_posts2020 = spark.read.format("ydb").options(my_ydb).load("stackoverflow/posts")
+ydb_posts2020: org.apache.spark.sql.DataFrame = [Id: bigint, PostTypeId: bigint ... 21 more fields]
+
+scala> ydb_posts2020.printSchema
+root
+ |-- Id: long (nullable = false)
+ |-- PostTypeId: long (nullable = true)
+ |-- AcceptedAnswerId: long (nullable = true)
+ |-- CreationDate: timestamp (nullable = true)
+ |-- Score: long (nullable = true)
+ |-- ViewCount: long (nullable = true)
+ |-- Body: binary (nullable = true)
+ |-- OwnerUserId: long (nullable = true)
+ |-- OwnerDisplayName: binary (nullable = true)
+ |-- LastEditorUserId: long (nullable = true)
+ |-- LastEditorDisplayName: binary (nullable = true)
+ |-- LastEditDate: timestamp (nullable = true)
+ |-- LastActivityDate: timestamp (nullable = true)
+ |-- Title: binary (nullable = true)
+ |-- Tags: binary (nullable = true)
+ |-- AnswerCount: long (nullable = true)
+ |-- CommentCount: long (nullable = true)
+ |-- FavoriteCount: long (nullable = true)
+ |-- ContentLicense: binary (nullable = true)
+ |-- ParentId: binary (nullable = true)
+ |-- CommunityOwnedDate: timestamp (nullable = true)
+ |-- ClosedDate: timestamp (nullable = true)
+ |-- Year: integer (nullable = true)
+
+scala> ydb_posts2020.count
+res3: Long = 4304021
+
+scala> ydb_posts2020.filter(col("AcceptedAnswerId") > 0).count
+res4: Long = 843780
+```
+
+## Spark-sql example {#example-spark-sql}
```

Comment:
This example does exactly the same thing as the previous one: let's merge them and provide commands for Shell and SQL in tabs. Ideally, doing the same with Scala/Java/Kotlin/etc. code would be helpful as well.